kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaEventListenerThread.java

```diff
@@ -0,0 +1,133 @@
+package org.hypertrace.core.kafka.event.listener;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigValue;
+import io.micrometer.core.instrument.Counter;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.InterruptException;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.hypertrace.core.serviceframework.metrics.PlatformMetricsRegistry;
+
+@Slf4j
+class KafkaEventListenerThread<K, V> extends Thread {
+  private static final String EVENT_CONSUMER_ERROR_COUNT = "event.consumer.error.count";
+  private static final String TOPIC_NAME = "topic.name";
+  private static final String POLL_TIMEOUT = "poll.timeout";
+  private final List<TopicPartition> topicPartitions;
+  private final Consumer<K, V> kafkaConsumer;
+  private final Duration pollTimeout;
+  private final Counter errorCounter;
+  private final List<BiConsumer<? super K, ? super V>> callbacks;
+
+  KafkaEventListenerThread(
+      String consumerName,
+      Config kafkaConfig,
+      Deserializer<K> keyDeserializer,
+      Deserializer<V> valueDeserializer,
+      List<BiConsumer<? super K, ? super V>> callbacks) {
+    this(
+        consumerName,
+        kafkaConfig,
+        new KafkaConsumer<>(
+            getKafkaConsumerConfigs(kafkaConfig.withFallback(getDefaultKafkaConsumerConfigs())),
+            keyDeserializer,
+            valueDeserializer),
+        callbacks);
+  }
+
+  KafkaEventListenerThread(
+      String consumerName,
+      Config kafkaConfig,
+      Consumer<K, V> kafkaConsumer,
+      List<BiConsumer<? super K, ? super V>> callbacks) {
+    super(consumerName);
+    this.setDaemon(true);
+    this.callbacks = callbacks;
+    this.pollTimeout =
+        kafkaConfig.hasPath(POLL_TIMEOUT)
+            ? kafkaConfig.getDuration(POLL_TIMEOUT)
+            : Duration.ofSeconds(30);
+    String topic = kafkaConfig.getString(TOPIC_NAME);
+    this.kafkaConsumer = kafkaConsumer;
+    List<PartitionInfo> partitions = kafkaConsumer.partitionsFor(topic);
+    topicPartitions =
+        partitions.stream()
```

Comment:
Can you get someone with more kafka context to review the kafka bits of this

Reply:
Sure, requesting @laxmanchekka to take a look. This code was a migration to HT, but good to review nevertheless

Reply:
Discussed offline. 

Need to document the following
- why we are using assign method instead of subsribing the whole topic at once?
- Why we are resetting the offsets to latest on every restart
- Rename the listener class (may be : KafkaLiveEventLister*)